doc/nrf/protocols/matter/getting_started/matter_api.rst

```diff
@@ -0,0 +1,248 @@
+
+.. code-block:: C++
+
+   #include "app/matter_init.h"
+
+   #include <zephyr/logging/log.h>
+
+   LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+   static void CustomMatterEventHandler(const ChipDeviceEvent *event, intptr_t /* unused */)
+   {
+     switch (event->Type) {
+     case DeviceEventType::kCHIPoBLEAdvertisingChange:
+     /* Custom code, e.g. control LED */
+     break;
+     }
+   }
+
+   CHIP_ERROR MatterAppInit()
+   {
+     /* Initialize Matter stack */
+     ReturnErrorOnFailure(Nrf::Matter::PrepareServer(Nrf::Matter::InitData{ .mPostServerInitClbk = [] {
+     LOG_INF("Matter server has been initialized.");
+     return CHIP_NO_ERROR;
+     } }));
+
+     /* Register custom Matter event handler. */
+     ReturnErrorOnFailure(Nrf::Matter::RegisterEventHandler(CustomMatterEventHandler, 0));
+
+     /* Application specific initialization, e.g. hardware initialization.
+
+     ...
+     */
+
+     /* Start Matter thread that will run the scheduled initialization procedure. */
+     return Nrf::Matter::StartServer();
+   }
+
+Note that the ``PrepareServer()`` call may contain more fields of the :c:struct:`InitData` being initialized.
+This function can be also called without any explicit argument. In that case, the default initialization will be provided.
+For more references and examples on how to leverage nRF Connect Matter APIs please reach :ref:`Matter samples <matter_samples>`' source code in nRF Connect SDK.
+
+Interacting with Matter Data Model
+**********************************
+
+Matter SDK Data Model interaction with the user's code is based on callbacks that can be implemented by the application.
+The generic callbacks that are common for Matter applications, regardless of the clusters configuration, are defined in :file:`ncs/modules/lib/matter/src/app/util/generic-callbacks.h` header file.
+The weak implementations of these functions, that can be overwritten in the application, are provided in :file:`ncs/modules/lib/matter/src/app/util/generic-callback-stubs.cpp` source file.
+For example, the :c:func:`MatterPostAttributeChangeCallback()` function is called by the Matter Data Model engine directly after an attribute value is changed.
+The value passed into this callback is the value to which the attribute was set by the framework.
+Apart from the value, this function is called with the attribute path ( of ``chip::app::ConcreteAttributePath``type) that can be used to filter the cluster and particular attribute of interest.
+:c:func:`MatterPostAttributeChangeCallback()` function is useful in case there is a need to provide the synchronization between the Data Model and the application state.
+For instance, the Matter device implementing a light bulb may drive the state of LED based on the ``On/Off`` attribute value.
+Every change of this attribute is reported by aforementioned callback and thus be captured in the application layer.
```

Comment:
```suggestion
Every change of this attribute is reported by the aforementioned callback and thus be captured in the application layer.

```
Note the added empty line.

Reply:
It should be "and thus can be ". Will fix.